Source/WTF/wtf/glib/GSpanExtras.h

```diff
@@ -21,9 +21,28 @@
 
 #include <wtf/StdLibExtras.h>
 #include <wtf/glib/GRefPtr.h>
+#include <wtf/glib/GUniquePtr.h>
 
 namespace WTF {
 
+template <typename T, size_t Extent = std::dynamic_extent>
+class GOwningSpan : public std::span<T> {
```

Comment:
Subclassing span looks risky here. Imagine someone allocating a GOwningSpan and storing it into a std::span variable (which is allowed due to inheritance). Now when the span gets destroyed, `~GOwningSpan` will not run (because the span destructor is not virtual) and will not free the memory.

Reply:
I think the lack of virtual destructor is reason enough to not do this.

It also seems quite confusing that it changes the normal ownership semantics of std::span.

Reply:
Good points, thanks both. After a bit of thinkging, I am now leaning towards:

- Rename the helper class to something else that does _not_ have “span” in its name. Not sure what name yet, but something like `GUniqueArrayPtr<T>`: Like a `GUniquePtr<T>` from the point of view that the class will _adopt_ a raw `T*` pointer and size (number of elements).
- Add a `GUniqueArrayPtr<T>::span() -> std::span<T>` function to it.
- Lastly, I am doubting whether to array operators to it like `operator[]` or not; maybe iterators are worth the effort?

I didn't see a way of creating something like a `VectorBuffer` that would allow adopting a raw pointer and customizing how memory is freed... I think we don't have something like this in WTF, right?

Reply:
This plan seems nicer than using `MallocPtr` since it clarifies that what we have is a buffer with an associated size.